Xamarin.Forms.Core/Shell/BaseShellItem.cs

```diff
@@ -112,6 +123,51 @@ public bool IsTabStop
 			set => SetValue(IsTabStopProperty, value);
 		}
 
+		public string BadgeText
+		{
+			get { return (string)GetValue(BadgeTextProperty); }
+			set { SetValue(BadgeTextProperty, value); }
+		}
+		public string BadgeMoreText => Shell.GetBadgeMoreText(this);
+
+		public Color BadgeTextColor => Shell.GetBadgeTextColor(this);
+
+		public Color BadgeUnselectedTextColor => Shell.GetBadgeTextColor(this);
+
+		public Color BadgeColor => Shell.GetBadgeColor(this);
+
+		public Color BadgeUnselectedColor => Shell.GetBadgeUnselectedColor(this);
+
+		public Color EffectiveBadgeTextColor => GetEffectiveBadgeTextColor(IsChecked);
+
+		public Color GetEffectiveBadgeTextColor(bool isSelected)
+		{
+			if (isSelected)
+			{
+				return Shell.Current.GetInheritedBadgeTextColor(this);
+			}
+
+			Color color = Shell.Current.GetInheritedBadgeUnselectedTextColor(this);
+
+			return !color.IsDefault ? color : Shell.Current.GetInheritedBadgeTextColor(this);
```

Comment:
We don't do inherited BP at this time (except for a11y). I'd prefer leaving this out for now, and come up with a proper way to do BP inheritance, than cook something custom every time we think we need it.

If you want fine-grained control over the colors, use a CSS stylesheet

Reply:
Dropped.